TELCODOCS-470 - Adding worker nodes to single-node OpenShift clusters using AI

aireilly opened the pull request:
Adds docs for adding worker nodes to single-node OpenShift clusters ([TELCODOCS-470](https://issues.redhat.com//browse/TELCODOCS-470)). Also clarifies installing SNO section. 

Version(s):
Merge to main, enterprise-4.11

Issue(s):
* https://issues.redhat.com/browse/TELCODOCS-470
* https://github.com/openshift/openshift-docs/issues/42214

Link to docs preview:
* TD-470: http://file.emea.redhat.com/aireilly/td-470/nodes/nodes/nodes-sno-worker-nodes.html
* GH#42214: http://file.emea.redhat.com/aireilly/td-470/installing/installing_sno/install-sno-installing-sno.html

omertuc commented on modules/sno-about-sno-cluster-worker-nodes.adoc:
```diff
@@ -0,0 +1,13 @@
+[id="sno-about-sno-cluster-worker-nodes_{context}"]
+= About worker nodes on single-node OpenShift clusters
+
+Worker nodes for single-node OpenShift clusters provide additional resources for your cluster. However, because there is only a single control plane node, ingress traffic is routed differently compared to standard cluster configurations with three control plane nodes.
```
```suggestion
Worker nodes for single-node OpenShift clusters provide additional resources for your cluster. However, because there is only a single control plane node, ingress traffic is, by default, routed differently compared to standard cluster configurations with three control plane nodes.
```

omertuc commented on modules/sno-about-sno-cluster-worker-nodes.adoc:
```diff
@@ -0,0 +1,13 @@
+
+Worker nodes for single-node OpenShift clusters provide additional resources for your cluster. However, because there is only a single control plane node, ingress traffic is routed differently compared to standard cluster configurations with three control plane nodes.
+
+In a default single-node cluster configuration, all ingress traffic is routed through the control plane node by default. One of the benefits of installing single-node clusters is the simplicity of not having to deal with load-balancing and virtual IPs. Not having a load balancer installed by default increases cost savings and simplifies cluster deployment and configuration.
```
>In a default single-node cluster configuration, all ingress traffic is routed through the control plane node by default.

This sentence feels a bit off. When you have just one node, it's trivial that all ingress traffic goes through it, because it has nowhere else to go.

When there is more than one node involved, it's less trivial to realize ingress traffic must flow the original SNO node, so that has to be mentioned

omertuc commented on modules/sno-about-sno-cluster-worker-nodes.adoc:
```diff
@@ -0,0 +1,13 @@
+
+In a default single-node cluster configuration, all ingress traffic is routed through the control plane node by default. One of the benefits of installing single-node clusters is the simplicity of not having to deal with load-balancing and virtual IPs. Not having a load balancer installed by default increases cost savings and simplifies cluster deployment and configuration.
+
+When you add a worker node to the cluster, the Assisted Installer sets the `DefaultPlacement` field of the `IngressController` CR to be `ControlPlane`. This effectively routes all ingress traffic to worker nodes through the control plane node by default.
```
>When you add a worker node to the cluster, the Assisted Installer sets the `DefaultPlacement` field of the `IngressController` CR to be `ControlPlane`. This effectively routes all ingress traffic to worker nodes through the control plane node by default.

nits:

- What's mentioned is a feature of the installer, not the assisted-installer

- defaultPlacement is a parameter of the ingress config CR, not the IngressController CR

I'm leaning towards omitting this entire paragraph, it's a bit of boring implementation details, as defaultPlacement is an immutable field users should not care about, it's just an internal way for the installer to communicate a general placement to the ingress controller

aireilly replied:
removed the paragraph.

omertuc commented on installing/installing_sno/install-sno-installing-sno.adoc:
```diff
@@ -6,16 +6,46 @@ include::_attributes/common-attributes.adoc[]
+== Installing single-node OpenShift using the Assisted Installer
 
-include::modules/install-sno-installing-with-usb-media.adoc[leveloffset=+1]
+To install OpenShift on a single node, first generate the installation ISO, then boot the server from the ISO. You can monitor the installation using the assisted installer.
```
This kinda sounds like the assisted installer is only there for monitoring, but it's much more than that. Also the term "installation ISO" is not entirely correct - in the context of the assisted installer, it's a discovery ISO.

Perhaps instead this paragraph should explain what the assisted installer is and how it can help you install single node openshift, something along the lines of "To install OpenShift on a single node, it's recommended to use the Assisted Installer, a simple web-based wizard that guides you and manages the installation"

omertuc commented on modules/install-sno-generating-the-discovery-iso-with-the-assisted-installer.adoc:
```diff
@@ -4,7 +4,7 @@
-= Generating the discovery ISO with the Assisted Installer
+= Generating the installation ISO with the Assisted Installer
 
 Installing {product-title} on a single node requires a discovery ISO, which the Assisted Installer (AI) can generate with the cluster name, base domain, Secure Shell (SSH) public key, and pull secret.
```
Even though these lines already existed - I'm not sure AI is an official acronym for the assisted installer, so perhaps it should be avoided. Also everything after and including "with the..." sounds a bit odd and too verbose. The users should go to the wizard, and the wizard will ask them for all those details. All those details also have nothing to do (at-least as far as the user is concerned) with the ISO file, so it's a bit odd to mention them in the same breath

omertuc commented on modules/install-sno-generating-the-discovery-iso-with-the-assisted-installer.adoc:
```diff
@@ -4,7 +4,7 @@
-= Generating the discovery ISO with the Assisted Installer
+= Generating the installation ISO with the Assisted Installer
 
 Installing {product-title} on a single node requires a discovery ISO, which the Assisted Installer (AI) can generate with the cluster name, base domain, Secure Shell (SSH) public key, and pull secret.
```
The instructions below this line are overly verbose and completely outdated. Button names change, titles change, and all of that happens completely independent of OCP releases - I think they should be removed - the wizard is very friendly and clear enough as it is, the user doesn't need a manual for this. 

Pretty much all we should do is remind them to tick the "Install single node OpenShift (SNO)" checkbox

omertuc replied:
There's also a section called "Monitoring the installation with the Assisted Installer" which should actually be "Start installation using the assisted installer" or something similar

It uses the term "Administration host" while originally it was "administration node" which is inconsistent

It doesn't mention that the user should wait for the host they're trying to discover using the discovery ISO to appear in the list of discovered hosts before proceeding with all those actions

omertuc replied:
Also "Monitoring the installation manually" - it's not monitoring, it's installing.

omertuc replied:
>Set the networkType. OVN-Kubernetes is the default networking for SNO. OpenShiftSDN is deprecated.

@romfreiman is it "deprecated" or completely disallowed?

romfreiman replied:
Yea. It was removed as a valid option from 4.11.

omertuc commented on modules/sno-worker-node-requirements.adoc:
```diff
@@ -0,0 +1,41 @@
+
+You can add worker nodes to single-node OpenShift clusters. Worker node hosts must meet the following requirements:
+
+* *Supported platforms:* Installing {product-title} on a single-node cluster worker node is supported on bare metal, vSphere, Red Hat OpenStack, and Red Hat Virtualization platforms. In all cases, you must specify the `platform.none: {}` parameter in the `install-config.yaml` configuration file.
```
>Installing {product-title} on a single-node cluster worker node is supported on bare metal

The grammar / meaning here is off

omertuc commented on modules/sno-about-sno-cluster-worker-nodes.adoc:
```diff
@@ -0,0 +1,9 @@
+[id="sno-about-sno-cluster-worker-nodes_{context}"]
+= About worker nodes on single-node OpenShift clusters
+
+Worker nodes for single-node OpenShift clusters provide additional resources for your cluster. However, because there is only a single control plane node, ingress traffic is, by default, routed through the control plane node. One of the benefits of installing single-node clusters is the simplicity of not having to deal with load-balancing and virtual IPs. Not having a load balancer installed by d ...
```
>However, because there is only a single control plane node, ingress traffic is, by default, routed through the control plane node.

This is not the reason we route traffic through the single node. The real reason is because before there were workers, there was only that node, and so all DNS entries / traffic was trivially pointed at it. And so we don't want users to have to change that once they add workers, so we by default keep routing all ingress traffic through the single node as we did originally.

I suggest something along the lines of:

```suggestion
Worker nodes for single-node OpenShift clusters provide additional resources for your cluster. Note that unlike multi-node clusters, by default all ingress traffic should still be routed to the single control-plane node, even after adding worker nodes.
```

aireilly commented on modules/install-sno-generating-the-discovery-iso-manually.adoc:
```diff
@@ -6,7 +6,11 @@
 = Generating the discovery ISO manually
 
-Installing {product-title} on a single node requires a discovery ISO, which you can generate with the following procedure.
+Installing {product-title} on a single node requires an discovery ISO, which you can generate with the following procedure.
```
@omertuc I swear I will get this if you say it enough times :)

romfreiman commented:
This part should be removed:
http://file.emea.redhat.com/aireilly/td-470/nodes/nodes/nodes-sno-worker-nodes.html#adding-worker-nodes-to-cluster_add-workers

omertuc commented:
>The users should not change their installed cluster to adding hosts cluster.

Yep, I believe @aireilly is aware, just didn't change it yet. @aireilly reach out if this needs clarification

lalon4 commented on modules/sno-adding-worker-nodes-to-sno-clusters.adoc:
```diff
@@ -0,0 +1,41 @@
+
+.Procedure
+
+. Log in to link:https://console.redhat.com/openshift/assisted-installer/clusters[console.redhat.com] and click on the single-node cluster that you want to add a worker node to.
```
should be https://console.redhat.com/openshift ? User cannot add hosts from the AI page

omertuc replied:
I agree, good catch.

aireilly replied:
Updated.

omertuc commented on modules/sno-adding-worker-nodes-to-sno-clusters.adoc:
```diff
@@ -0,0 +1,41 @@
+
+. As the installation proceeds, the installation generates pending certificate signing requests (CSRs) for the worker node. When prompted, approve the pending CSRs to complete the installation.
++
+When the worker node is sucessfully installed, it is listed under the single-node cluster on link:https://console.redhat.com/openshift/assisted-installer/clusters[console.redhat.com].
```
I agree, good catch.

Amrita42 commented on installing/installing_sno/install-sno-installing-sno.adoc:
```diff
@@ -6,16 +6,42 @@ include::_attributes/common-attributes.adoc[]
+[role="_additional-resources"]
+.Additional resources
+
+* For more information about creating a bootable USB drive, see xref:../../installing/installing_sno/install-sno-installing-sno.adoc#installing-with-usb-media_install-sno-installing-sno-with-the-assisted-installer[Creating a bootable ISO image on a USB drive].
```
Since its mentioned in the square brackets, I suggest:

_- xref:../../installing/installing_sno/install-sno-installing-sno.adoc#installing-with-usb-media_install-sno-installing-sno-with-the-assisted-installer[Creating a bootable ISO image on a USB drive]_

Amrita42 commented on nodes/nodes/nodes-sno-worker-nodes.adoc:
```diff
@@ -0,0 +1,69 @@
+
+== Adding worker nodes to {sno} clusters
+
+There are several ways that you can add worker nodes to a single-node cluster. You can add worker nodes on link:https://console.redhat.com/openshift/assisted-installer/clusters[console.redhat.com], you can add worker nodes to clusters manually, or you can use the Assisted Installer REST API directly.
```
Suggestion >> How about a list for example>> 
 There are several ways that you can add worker nodes to a single-node cluster. You can add worker nodes :
* link:https://console.redhat.com/openshift/assisted-installer/clusters[console.redhat.com]
* Manually
* Directly use Assisted Installer REST API

lalon4 replied:
should be: link:https://console.redhat.com/openshift/[console.redhat.com]

Amrita42 commented on nodes/nodes/nodes-sno-worker-nodes.adoc:
```diff
@@ -0,0 +1,69 @@
+
+=== Adding worker nodes using the Assisted Installer API
+
+You can add worker nodes to clusters using the Assisted Installer API. Before you can add worker nodes, you need to install the OpenShift Cluster Manager CLI (`ocm`) and authenticate against the API.
```
As per [ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n)>> Use "need to" only when "must" is too restrictive. Suggest using **must** in this context.

omertuc replied:
I would argue that `must` is too restrictive in this particular case, because you technically could authenticate against the API in ways other than using the OCM CLI

Amrita42 replied:
ok noted then please ignore this review comment.

Amrita42 replied:
cc @aireilly

Amrita42 commented on nodes/nodes/nodes-sno-worker-nodes.adoc:
```diff
@@ -0,0 +1,69 @@
+[role="_additional-resources"]
+.Additional resources
+
+* During installation, you need to approve certificate signing requests for the worker node. For more information, see xref:../../nodes/nodes/nodes-sno-worker-nodes.adoc#installation-approve-csrs_add-workers[Approving the certificate signing requests for your machines].
```
As per [ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n)>> Use "need to" only when "must" is too restrictive. Suggest using **must** in this context.

Amrita42 commented on nodes/nodes/nodes-sno-worker-nodes.adoc:
```diff
@@ -0,0 +1,69 @@
+[role="_additional-resources"]
+.Additional resources
+
+* During installation, you need to approve certificate signing requests for the worker node. For more information, see xref:../../nodes/nodes/nodes-sno-worker-nodes.adoc#installation-approve-csrs_add-workers[Approving the certificate signing requests for your machines].
```
As per [ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n)>> Use "need to" only when "must" is too restrictive. Suggest using **must** in this context.

Amrita42 commented on nodes/nodes/nodes-sno-worker-nodes.adoc:
```diff
@@ -0,0 +1,69 @@
+[role="_additional-resources"]
+.Additional resources
+
+* For more information about worker node hardware requirements, see xref:../../installing/installing_bare_metal/installing-restricted-networks-bare-metal.adoc#installation-minimum-resource-requirements_installing-restricted-networks-bare-metal[Minimum resource requirements for cluster installation].
```
I would suggest for all the xrefs , to use a similar format as per https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections.  For example

- xref:../../installing/installing_bare_metal/installing-restricted-networks-bare-metal.adoc#installation-minimum-resource-requirements_installing-restricted-networks-bare-metal[Minimum resource requirements for cluster installation].

Amrita42 commented on nodes/nodes/nodes-sno-worker-nodes.adoc:
```diff
@@ -0,0 +1,69 @@
+[role="_additional-resources"]
+.Additional resources
+
+* During installation, you need to approve certificate signing requests for the worker node. For more information, see xref:../../nodes/nodes/nodes-sno-worker-nodes.adoc#installation-approve-csrs_add-workers[Approving the certificate signing requests for your machines].
```
As per [ISG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n)>> Use "need to" only when "must" is too restrictive. Suggest using must in this context

lalon4 commented on modules/sno-adding-worker-nodes-to-sno-clusters.adoc:
```diff
@@ -0,0 +1,41 @@
+[id="sno-adding-worker-nodes-to-sno-clusters_{context}"]
+= Adding worker nodes using the Assisted Installer on console.redhat.com
+
+You can add worker nodes to {sno} clusters that were created on link:https://console.redhat.com[console.redhat.com] using the Assisted Installer.
```
Suggesting to add the link to Assisted Installer: link:https://console.redhat.com/openshift/assisted-installer/clusters/~new[Assisted Installer]

lalon4 commented on modules/ai-adding-worker-nodes-to-cluster.adoc:
```diff
@@ -0,0 +1,303 @@
+  -d '{
+  "api_vip_dnsname": "<api_vip>", <1>
+  "openshift_cluster_id": "00000000-0000-0000-0000-000000000000", <2>
+  "name": "openshift-cluster-name",
```
should be without the comma: `"name": "openshift-cluster-name"`

lalon4 commented on modules/ai-adding-worker-nodes-to-cluster.adoc:
```diff
@@ -0,0 +1,303 @@
+  "pull_secret": $pull_secret[0] | tojson,
+  "cluster_id": $cluster_id,
+  "ssh_authorized_key": $ssh_pub_key,
+  "image_type": <iso_image_type>
```
suggesting: `"image_type": "<iso_image_type>"`

lalon4 commented on modules/ai-adding-worker-nodes-to-cluster.adoc:
```diff
@@ -0,0 +1,303 @@
++
+[source,terminal]
+----
+$ curl -L "https://api.openshift.com/api/assisted-images/images/41b91e72-c33e-42ee-b80f-b5c5bbf6431a?arch=x86_64&image_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NTYwMjYzNzEsInN1YiI6IjQxYjkxZTcyLWMzM2UtNDJlZS1iODBmLWI1YzViYmY2NDMxYSJ9.1EX_VGaMNejMhrAvVRBS7PDPIQtbOOc8LtG8OukE1a4&type=minimal-iso&version=4.11" --output rhcos-live-minimal.iso
```
suggesting to add -s flag to avoid download progress prints
i.e 
curl -L -s "https://api...."

lalon4 commented:
/lgtm on add worker nodes
Still need to review installing-sno

achuzhoy commented on modules/sno-adding-worker-nodes-to-sno-clusters-manually.adoc:
```diff
@@ -0,0 +1,110 @@
+
+.Prerequisites
+
+* Install a cluster on bare metal.
```
a cluster is usually a collection, so we might want to emphasize SNO

lpettyjo commented on modules/ai-sno-requirements-for-installing-worker-nodes.adoc:
```diff
@@ -0,0 +1,49 @@
++
+The server must have a Baseboard Management Controller (BMC) when booting with virtual media.
+
+* *Networking:* The worker node server must have access to the internet or access to a local registry if it is not connected to a routable network. The worker node server must have a DHCP reservation or a static IP address and be able to access the {sno} cluster Kubernetes API, Ingress route, and cluster node domain names. You must configure the DNS to resolve the IP address to each of the follow ...
```
I don't think that "Ingress route" should be capitalized.

omertuc replied:
Looks like it's copied from https://github.com/openshift/openshift-docs/blob/20ace352d73717d91949f7e05e223b445649f3e5/modules/install-sno-requirements-for-installing-on-a-single-node.adoc?plain=1#L32

omertuc replied:
(not to say that it should not be fixed, just a remark)

aireilly replied:
Fixed in both places

lpettyjo commented on modules/cli-installing-ocm.adoc:
```diff
@@ -0,0 +1,93 @@
+
+.Procedure
+
+. Navigate to link:https://github.com/openshift-online/ocm-cli/releases/[github.com/openshift-online/ocm-cli/releases] and choose the asset for your operating system and architecture.
```
"Go to" instead of "Navigate". IBM SG.

lpettyjo commented on modules/install-sno-installing-with-the-assisted-installer.adoc:
```diff
@@ -0,0 +1,29 @@
+
+. Attach the {op-system} discovery ISO to the target host.
+
+. Reboot the server and enter the BIOS settings upon reboot.
```
Use "after" instead of "upon".

lpettyjo commented on modules/install-sno-installing-with-the-assisted-installer.adoc:
```diff
@@ -0,0 +1,29 @@
+
+. Remove the discovery ISO, and reset the BIOS to boot to the server's newly installed hard disk.
++
+The server will restart several times, deploying the control plane.
```
Change to present tense: "The server restarts several times,..."

lpettyjo commented on modules/sno-adding-worker-nodes-to-sno-clusters.adoc:
```diff
@@ -0,0 +1,36 @@
+
+. Click *Add hosts*, and download the discovery ISO for the new worker node, adding SSH public key and configuring cluster-wide proxy settings as required.
+
+. Boot the target host using the discovery ISO, and wait for the host to be discovered in the console. Once the host is discovered, start the install.
```
Use "After" rather than "Once".

lpettyjo commented on nodes/nodes/nodes-sno-worker-nodes.adoc:
```diff
@@ -0,0 +1,82 @@
+
+There are several ways that you can add worker nodes to a single-node cluster. You can add worker nodes to a cluster manually, using link:https://console.redhat.com/openshift/assisted-installer/clusters[console.redhat.com], or by using the Assisted Installer REST API directly.
+
+Adding worker nodes does not expand the cluster control plane, and it does not provide high availability to your cluster. For {sno} clusters, high availability is handled by failing over to another site. It is not recommended to add a large number of worker nodes to a single-node cluster.
```
Might want to make this a note or important note.

lpettyjo reviewed (commented):
Otherwise, LGTM! Nice work!

omertuc commented on modules/ai-adding-worker-nodes-to-cluster-on-prem.adoc:
```diff
@@ -0,0 +1,325 @@
+
+[WARNING]
+====
+On-premises Assisted Installer REST API deployments are not supported by Red Hat.
```
This is temporary, it's just for a particular customer, it will be compiled for them & removed before we merge

lalon4 commented:
/lgtm

omertuc commented on modules/install-sno-installing-with-the-assisted-installer.adoc:
```diff
@@ -0,0 +1,25 @@
+
+. Complete the install wizard steps. Add networking details, including a subnet from the available subnets. Add the SSH public key if necessary.
+
+. Monitor the installation's progress. Watch the cluster events. After the installation process finishes writing the discovery image to the server's drive, the server restarts.
```
```suggestion
. Monitor the installation's progress. Watch the cluster events. After the installation process finishes writing the operating system image to the server's drive, the server restarts.
```

omertuc commented on modules/install-sno-installing-with-the-assisted-installer.adoc:
```diff
@@ -0,0 +1,25 @@
+
+. Remove the discovery ISO, and reset the BIOS to boot to the server's newly installed hard disk.
++
+The server restarts several times, deploying the control plane.
```
```suggestion
The server may restart several times, deploying the control plane.
```

To emphasize it's an automatic process

aireilly replied:
This wording doesn't imply an automatic process. it implies that there is a possibility that the server will restart.

aireilly replied:
"The server restarts several times automatically, deploying the control plane."

omertuc commented on modules/install-sno-installing-with-usb-media.adoc:
```diff
@@ -4,34 +4,26 @@
+= Creating a bootable ISO image on a USB drive
 
-Installing with USB media involves creating a bootable USB drive with the discovery ISO on the administration node. Booting the server with the USB drive prepares the node for a single node installation.
+You can install the {sno} cluster using a bootable USB drive that contains the ISO image. Booting the server with the USB drive prepares the node for a single node installation.
```
I think we should remove the word OpenShift from this paragraph and maybe change the wording a little bit to clarify that it's just generic, non-openshift related section on how to boot ISO files from USB drives

lalon4 commented:
/lgtm

kalexand-rh commented on modules/install-sno-monitoring-the-installation-manually.adoc:
```diff
@@ -4,22 +4,28 @@
+= Monitoring the cluster installation using openshift-install
 
-If you created the ISO manually, use this procedure to monitor the installation.
+Use `openshift-install` to monitor the progress of the single-node cluster install.
```
> Do not use a word as a part of speech that is different from its standard dictionary definition. In particular, do not use verbs as nouns or adjectives; examples are configure, compile, debug, fix, and install.

You have several questionable instances of "install" in this PR.

aireilly replied:
fixed

kalexand-rh commented on modules/sno-adding-worker-nodes-to-sno-clusters-manually.adoc:
```diff
@@ -0,0 +1,110 @@
+$ ISO_URL=$(./openshift-install coreos print-stream-json | grep location | grep $ARCH | grep iso | cut -d\" -f4)
+----
+
+. Use the RHCOS ISO and the hosted `worker.ign` file to install the worker node:
```
You should expand RHCOS on first use, and {op-system-first} is the attribute you want for it.

kalexand-rh commented on modules/sno-adding-worker-nodes-to-sno-clusters-manually.adoc:
```diff
@@ -0,0 +1,110 @@
+
+. Use the RHCOS ISO and the hosted `worker.ign` file to install the worker node:
+
+.. Boot the target host using the RHCOS ISO and your preferred method of installation.
```
Other RHCOS instances are {op-system}.

kalexand-rh commented on modules/sno-adding-worker-nodes-to-sno-clusters-manually.adoc:
```diff
@@ -0,0 +1,110 @@
+<hard_disk>:: is hard disk where you install {op-system}, for example, `/dev/sda`.
+--
+
+. As the installation proceeds, the installation generates pending certificate signing requests (CSRs) for the worker node. When prompted by installer, approve the pending CSRs to complete the installation.
```
```suggestion
. As the installation proceeds, the installation generates pending certificate signing requests (CSRs) for the worker node. When prompted by installation program, approve the pending CSRs to complete the installation.
```

kalexand-rh commented on modules/sno-adding-worker-nodes-to-sno-clusters.adoc:
```diff
@@ -0,0 +1,36 @@
+
+.Procedure
+
+. Log in to link:https://console.redhat.com/openshift[console.redhat.com] and click on the single-node cluster that you want to add a worker node to.
```
```suggestion
. Log in to link:https://console.redhat.com/openshift[console.redhat.com] and click the single-node cluster that you want to add a worker node to.
```

kalexand-rh reviewed (commented):
This is very close. I see a few small format issues, but my big question is whether "console.redhat.com" or OpenShift Cluster Manager is correct.

aireilly commented on modules/ai-authenticating-against-ai-rest-api.adoc:
```diff
@@ -0,0 +1,72 @@
+
+* Log in to link:https://console.redhat.com/openshift/assisted-installer/clusters[console.redhat.com] as a user with cluster creation privileges.
+
+* Install `jq`.
```
We're dealing with a REST API (not oc) and jq is absolutely a reasonable requirement for a sys admin/developer in this context.

kalexand-rh replied:
This was a request from support because jq is not available on RHEL 7.

aireilly replied:
Ok interesting. In general I follow this guidance in my docs. This was an exception. I'm going to look a bit further into this. Thanks.

aireilly replied:
OK. I verified that `jq` _is_ at least available by default on RHEL 8. However, I do want to proceed with this "install `jq`" directive as is. In this instance, it is "truly required". Also, this is a very high priority epic for Telco in 4.11. I've already peeled off an unsupported on-prem procedure to a KCS article for this epic, I really want to keep this one in place in core docs. Let me know your thoughts @kalexand-rh thanks.

kalexand-rh replied:
If it's truly required, that is fine! It's a good thing to check in on - sometimes the commands can be reworked to avoid that package. I'm glad that they've made the package available in RHEL 8 and that someday this won't be an issue anymore.

aireilly commented on modules/cli-installing-ocm.adoc:
```diff
@@ -0,0 +1,93 @@
+
+.Procedure
+
+. Go to link:https://github.com/openshift-online/ocm-cli/releases/[github.com/openshift-online/ocm-cli/releases] and choose the asset for your operating system and architecture.
```
Oop. This module shouldn't be in PR. It's removed from assembly I forgot to delete. Will remove.

aireilly commented:
> This is very close. I see a few small format issues, but my big question is whether "console.redhat.com" or OpenShift Cluster Manager is correct.

Thanks! Will investigate :)

aireilly commented on modules/ai-adding-worker-nodes-to-cluster.adoc:
```diff
@@ -0,0 +1,340 @@
+where:
++
+--
+<api_vip> :: is the hostname for the cluster's API server. This can be the DNS domain for the API server or the IP address of the single node which the worker node can reach. For example, `api.compute-1.example.com`.
```
Updated.

aireilly commented on modules/ai-authenticating-against-ai-rest-api.adoc:
```diff
@@ -0,0 +1,72 @@
+//
+// * nodes/nodes/nodes-sno-worker-nodes.adoc
+
+:_module-type: PROCEDURE
```
Corrected in 4 places.

aireilly commented:
I've addressed your comments @kalexand-rh. I adjusted references to console.redhat to {cluster-manager} and applied other suggestions thanks. Provided you are happy to leave `jq` references in place, should be good to merge provided there is nothing else.

mburke5678 commented:
/cherrypick enterprise-4.11
